### PR TITLE
Update frequenz-api-common to v0.5.0

### PR DESCRIPTION
**Pull request by tiyash-basu-frequenz**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -63,6 +63,11 @@
   `frequenz.api.microgrid.v1.<package>`. `v1` is the API's major version, and
   will be incremented for breaking changes.
 
+- The common protobuf dependency has been upgraded to `v0.5.0`. The protobuf
+  messages returned by the RPCs are now in the `frequenz.api.common.v1` package.
+  As a result, all files besides `microgrid.proto` have become obsolete, and
+  should be removed.
```
Comment (llucax) [inline RELEASE_NOTES.md]: :scissors: :tada:

File: pyproject.toml [hunk 2]
```diff
@@ -26,7 +26,7 @@ classifiers = [
 ]
 requires-python = ">= 3.11, < 4"
 dependencies = [
-  "frequenz-api-common >= 0.4.0, < 0.5",
+  "frequenz-api-common@git+https://github.com/frequenz-floss/frequenz-api-common",
```
Comment (llucax) [inline pyproject.toml]: I guess this can be put back to point to a proper release  now that 0.5.0 is released.

**Review by llucax — COMMENTED** [review]
You probably want to also update [the `mkdocs.yml` file](https://github.com/frequenz-floss/frequenz-api-microgrid/blob/e4323472fe032f4846fc914b7077291cb3a5d858/mkdocs.yml#L109) (maybe also do a `git grep api-common` to make sure we are not missing any other reference?

File: RELEASE_NOTES.md (hunk 1 above)
Comment (tiyash-basu-frequenz) [reply RELEASE_NOTES.md]: I actually updated the note. The files _have been_ removed.

File: pyproject.toml (hunk 2 above)
Comment (tiyash-basu-frequenz) [reply pyproject.toml]: Oops, had forgotten about this. Thanks for pointing out. Fixed.

File: RELEASE_NOTES.md (hunk 1 above)
Comment (llucax) [reply RELEASE_NOTES.md]: Haha, I got so excited I didn't read right through the end :rofl:

**Comment by llucax** [comment]
I guess you didn't saw the review comment above?

image

GitHub should really do something about those review comments, they are very easy to miss and I just realized they don't even have a way to put a link to them!

**Comment by tiyash-basu-frequenz** [comment]
>I guess you didn't saw the review comment above?

You are right, I missed it.

It was a good suggestion though. I fixed the line you had linked. Plus, I also found another line in the release notes that referred to v0.3, and removed it.

**Review by llucax — APPROVED** [review]

**Pull request merged**